autoload/rhubarb.vim

```diff
@@ -268,7 +268,26 @@ function! rhubarb#FugitiveUrl(opts, ...) abort
   else
     let commit = a:opts.commit
   endif
-  if get(a:opts, 'type', '') ==# 'tree' || a:opts.path =~# '/$'
+  if get(a:opts, 'type', '') ==# 'commit'
+    let url = root . '/commit/' . commit
+    let start = get(a:opts, 'start')
+    let end = get(a:opts, 'end')
+    if !empty(start) && !empty(end) && start.oldpath ==# end.oldpath && start.newpath ==# end.newpath && executable('md5')
+      let url .= '#diff-' . system('md5 -qs ' . s:shellesc(len(start.newpath) > 0 ? start.newpath : start.oldpath))[0:-2]
```

Comment:
`md5` is a macOS thing, at a bare minimum we need to support `md5sum` as well. A bit of googling suggests [CertUtil](https://superuser.com/questions/245775/is-there-a-built-in-checksum-utility-on-windows-7/898377#898377) might work for Windows.

(Why on earth did GitHub choose to use MD5 when SHA-1 is a drop-in equivalent and part of Git itself?)

Reply:
It looks like they're using SHA-256 now? Vim can do that natively with `sha256()` so at least that's one problem solved.

Reply:
nice